compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -831,47 +820,154 @@ impl InitMask {
     }
 }
 
+/// A contiguous chunk of initialized or uninitialized memory.
+pub enum InitChunk {
+    Init(Range<Size>),
+    Uninit(Range<Size>),
+}
+
+impl InitChunk {
+    #[inline]
+    pub fn range(&self) -> Range<Size> {
+        match self {
+            Self::Init(r) => r.clone(),
+            Self::Uninit(r) => r.clone(),
+        }
+    }
+}
+
 /// Yields [`InitChunk`]s. See [`InitMask::range_as_init_chunks`].
 pub struct InitChunkIter<'a> {
     init_mask: &'a InitMask,
+    /// Whether the last chunk was initialized.
```

Comment:
"previous" would be more clear than "last", I think.

But also, there might not be a previous chunk -- what does this flag mean initially?

Reply:
Hmm this comment is actually backwards--should be the next chunk, not the previous